jsonrpc/server.go

```diff
@@ -53,6 +53,13 @@ type Error struct {
 	Data    any    `json:"data,omitempty"`
 }
 
+// DuplicateWithData copies the error and sets the data field on the copy
+func (e *Error) DuplicateWithData(data any) *Error {
```

Comment:
nit: Consider renaming it to `AddData()`

Reply:
I want to emphasize that it copies the error before adding the data. Will rename it to `CloneWithData()`